lib/lambda.js

```diff
@@ -86,11 +87,14 @@ module.exports = function elasticApmAwsLambda (agent) {
              * elastic-apm-traceparent if available.
              */
             parentId = value
```

Comment:
1. The handling will have to change to avoid that `break` preferring 'elastic-apm-traceparent' to 'traceparent'. IIUC, with the current logic it will miss tracestate if it comes after 'elastic-apm-traceparent' in `payload.headers`.
2. Also, per #2079 the logic should change (could be a separate PR) to prefer 'traceparent' to 'elastic-apm-traceparent'.

Reply:
@trentm Ugh -- that's a weird construct.  Thanks for catching it. Any strong preference between 

- more code and fewer loop cycles (matching most closely with current intent/pattern)

```
    if (payload.headers !== undefined) {
      for (const [key, value] of Object.entries(payload.headers)) {
        const lowerCaseKey = key.toLowerCase()
        if (lowerCaseKey === 'traceparent') {
          parentId = value
        } else if (!parentId && lowerCaseKey === 'elastic-apm-traceparent') {
          parentId = value
        } else if(lowerCaseKey === 'tracestate') {
          tracestate = value
        }
        if(parentId && tracestate) {
          break
        }
      }
    }
```

- or less code but slightly more work normalizing all the headers

```
      if (payload.headers !== undefined) {
        const normalizedHeaders = {}
        for (const [key, value] of Object.entries(payload.headers)) {
          const lowerCaseKey = key.toLowerCase()
          normalizedHeaders[lowerCaseKey] = value
        }
        parentId = normalizedHeaders.traceparent ? normalizedHeaders.traceparent : normalizedHeaders['elastic-apm-traceparent']
        tracestate = normalizedHeaders.tracestate
      }
```

Reply:
The former will prefer 'elastic-apm-traceparent' over 'traceparent' if it and 'tracestate' come first, so doesn't work as written.

No strong preference between those two until perf profiles show that the section is a hot spot.

Slightly related: from old https://elastic.slack.com/archives/C5M02D1GE/p1619802623086700?thread_ts=1619742610.083400&cid=C5M02D1GE `for ([key, value] of Object.entries(obj))` is dog slow. Perhaps premature optimization, but I'd generally bias against using it.